packages/ccd-js-gen/README.md

```diff
@@ -0,0 +1,484 @@
+module `./my-contract.wasm.v1`:
+
+```
+ccd-js-gen --module ./my-contract.wasm.v1 --out-dir ./generated
+```
+
+> For a dapp project, it is recommended to have this as part of a script in `package.json`.
+
+## Using the library
+
+This package can be used programmatically as well.
+
+```typescript
+import * as ccdJsGen from "@concordium/ccd-js-gen"
+```
+
+### Generate from smart contract module file
+
+To generate smart contract clients for a smart contract module file,
+either downloaded from the blockchain or build from the smart contract source code:
+
+```typescript
+import * as ccdJsGen from "@concordium/ccd-js-gen"
+
+const moduleFilePath = "./my-contract.wasm.v1"; // Path to smart contract module.
+const outDirPath = "./generated"; // The directory to use for the generated files.
+
+// Read the module and generate the smart contract clients.
+console.log('Generating smart contract module clients.')
+await ccdJsGen.generateContractClientsFromFile(moduleFilePath, outDirPath);
+console.log('Code generation was successful.')
+```
+
+### Generate from smart contract module on chain
+
+To generate smart contract clients for a smart contract module on chain, you need access to the gRPC of a Concordium node.
+Use `@concordium/web-sdk` to download the smart contract module and use it to generate the clients.
+
+```typescript
+import * as ccdJsGen from "@concordium/ccd-js-gen"
+import * as SDK from "@concordium/web-sdk"
+
+const moduleFilePath = "./my-contract.wasm.v1"; // Path to smart contract module.
+const outDirPath = "./generated"; // The directory to use for the generated files.
+const outputModuleName = "wCCD-module"; // The name to give the output smart contract module.
+
+const grpcClient = ...; // A Concordium gRPC client from @concordium/web-sdk.
+const moduleRef = SDK.ModuleReference.fromHexString('<hex of module referecene>');
+
+// Fetch the smart contract module source from chain.
+const moduleSource = await grpcClient.getModuleSource(moduleRef);
+
+// Generate the smart contract clients from module source.
+console.info('Generating smart contract module clients.');
+await ccdJsGen.generateContractClients(moduleSource, outputModuleName, outDirPath);
+console.info('Code generation was successful.');
+```
+
+## Using the generated client
+
+The generator produces a file with functions for interacting with the smart contract module and files for each smart contract in the smart contract module used.
+
+For example: generating clients for a smart contract module `my-module` containing
+the smart contracts `my-contract-a` and `my-contract-b`.
+into the directory `./generated` produces the following structure:
+
+```
+generated/
+├─ my-module.js        // Functions for interacting with the 'my-module' smart contract module on chain.
+├─ my-contract-a.js    // Functions for interacting with the 'my-contract-a' smart contract.
+└─ my-contract-b.js    // Functions for interacting with the 'my-contract-b' smart contract.
```

Comment:
I did not notice that before, but is name conflict handled?

If I call my module `cis2` and it has a single contract `cis2` will that work?

Reply:
I have changed this to include the module name as prefix for the contracts